integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,910 @@
+    } else if (Integration.options.gaOptions.enhancedEcommerce) {
+      Integration.productListViewed = Integration.productListViewedEnhanced;
+      Integration.productClicked = Integration.productClickedEnhanced;
+      Integration.productViewed = Integration.productViewedEnhanced;
+      Integration.productAdded = Integration.productAddedEnhanced;
+      Integration.productRemoved = Integration.productRemovedEnhanced;
+      Integration.promotionViewed = Integration.promotionViewedEnhanced;
+      Integration.promotionClicked = Integration.promotionClickedEnhanced;
+      Integration.checkoutStarted = Integration.checkoutStartedEnhanced;
+      Integration.orderRefunded = Integration.orderRefundedEnhanced;
+      Integration.orderCompleted = Integration.orderCompletedEnhanced;
+      Integration.checkoutStepViewed = Integration.checkoutStepViewedEnhanced;
+      Integration.checkoutStepCompleted =
+        Integration.checkoutStepCompletedEnhanced;
+      Integration.orderUpdated = Integration.orderUpdatedEnhanced;
+
+      // Additional event on top og GA destination
+      Integration.productAddedToWishlist = Integration.productAddedToWishlistEnhanced;
+      Integration.productShared = Integration.productSharedEnhanced;
+      Integration.productsSearched = Integration.productsSearchedEnhanced;
+
+      // This mapping is for events which are not supported by segment and will
+      // be tracked by track method.
+      Integration.customEventsMapping = {
+        login: Integration.loggedInEnhanced,
+        sign_up: Integration.signedUpEnhanced,
+        exception: Integration.exceptionOccuredEnhanced,
+        timing_complete: Integration.timingCompletedEnhanced,
+        generate_lead: Integration.leadGeneratedEnhanced,
+        set_checkout_option: Integration.setCheckoutOptionEnhanced
+      };
+    }
+    /* eslint-enable */
+});
+
+/**
+ * Initialize.
+ *
+ * https://developers.google.com/gtagjs
+ *
+ * @api public
+ */
+
+GTAG.prototype.initialize = function() {
+  var config = [];
+  var that = this;
+  var gaOptions = this.options.gaOptions;
+  var gaWebMeasurementId = this.options.gaWebMeasurementId;
+  var gaWebAppMeasurementId = this.options.gaWebAppMeasurementId;
+  var awConversionId = this.options.awConversionId;
+  var dcFloodLightId = this.options.dcFloodLightId;
+  var accountId =
+    gaWebMeasurementId ||
+    gaWebAppMeasurementId ||
+    awConversionId ||
+    dcFloodLightId;
+  if (gaWebMeasurementId || gaWebAppMeasurementId) {
+    var gaSetting = {};
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      // To Set persistent values we need to use set instead of config
+      // https://developers.google.com/analytics/devguides/collection/gtagjs/setting-values
+
+      gaSetting.custom_map = merge(
+        gaOptions.dimensions,
+        gaOptions.metrics,
+        gaOptions.contentGroupings
```

Comment:
I believe `contentGroupings` should not be nested under the `custom_map`. I believe `custom_map` is just for custom dimensions and metrics, if I remember correctly. These are the docs for content groupings: https://support.google.com/analytics/answer/7475939?hl=en#code

Reply:
@briemcnally
Yup, you are right, thanks for clarification.